app/controllers/bulk_downloads_controller.rb

```diff
@@ -2,8 +2,55 @@
 class BulkDownloadsController < ApplicationController
   include BulkDownloadTypesHelper
 
+  before_action do
+    allowed_feature_required("bulk_downloads")
+  end
+
   # GET /bulk_downloads/types
   def types
     render json: BulkDownloadTypesHelper::BULK_DOWNLOAD_TYPES
   end
+
+  # POST /bulk_downloads
+  def create
+    bulk_download_params = params.permit(:download_type, sample_ids: [], params: {})
+
+    # Convert sample ids to pipeline run ids.
+    pipeline_run_ids = Sample.where(id: bulk_download_params[:sample_ids]).includes(:pipeline_runs)
```

Comment:
You'll want to read from power here.

Reply:
Also, we should probably create a method `first_pipeline_runs` or something to do this in PipelineRun model that will do basically something like:
```
def first_pipeline_runs(samples)
	PipelineRun.where(samples: samples).group(:sample_id).having('MAX(`pipeline_runs`.`created_at`) >= `pipeline_runs`.`created_at`')
end
```
This is a very common use case... 
We can also add your conditions in there.

Reply:
Looks like the function already exists. Will use. Thanks!